File: 0501-find-mode-in-binary-search-tree/0501-find-mode-in-binary-search-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def findMode(self, root: Optional[TreeNode]) -> List[int]:
        mp = defaultdict(int)
        q = deque([root])
        while q:
            node = q.popleft()
            mp[node.val] += 1
            if node.left:
                q.append(node.left)
            if node.right:
                q.append(node.right)
        mxFreq = max(mp.values())
        ans = []
        for key in mp:
            if mp[key] == mxFreq:
                ans.append(key)
        return ans

File: 0501-find-mode-in-binary-search-tree/0501-find-mode-in-binary-search-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> findMode(TreeNode* root) {
        unordered_map<int, int> mp;
        queue<TreeNode*> q;
        q.push(root);
        while(!q.empty()){
            TreeNode *node = q.front(); q.pop();
            mp[node->val]++;
            if(node->left) q.push(node->left);
            if(node->right) q.push(node->right);
        }
        int mxFreq = 0;
        for(auto [key, val] : mp) mxFreq = max(mxFreq, val);
        vector<int> ans;
        for(auto [key, val] : mp){
            if(mxFreq == val) ans.push_back(key);
        }
        return ans;
    }
};